pandas/core/indexes/base.py

```diff
@@ -2489,8 +2489,12 @@ def _union(self, other, sort):
                 value_set = set(lvals)
                 result.extend([x for x in rvals if x not in value_set])
         else:
-            indexer = self.get_indexer(other)
-            indexer, = (indexer == -1).nonzero()
+            # find indexes of things in "other" that are not in "self"
+            if self.is_unique:
+                indexer = self.get_indexer(other)
+                indexer, = (indexer == -1).nonzero()
```

Comment:
i dont think we use the trailing comma pattern often.  Can you use `[0]` instead?

Reply:
I think this is a bad change because not only is your suggested alternative longer, but it also removes a check that the 2nd dimension is a singleton. I'm not sure that your argument for changing it is very compelling. But I don't really want to argue about it, so I'll change it!